syntax = "proto3";

message VideoMessage {
  uint32 width = 1; // 视频宽度
  uint32 height = 2; // 视频高度
  uint32 frame_rate = 3; // 视频帧率
  uint32 bit_rate = 4; // 视频比特率
  bytes data = 5; // 视频数据
}

message MouseMessage {
  uint32 x = 1; // 鼠标x坐标
  uint32 y = 2; // 鼠标y坐标
  uint32 button = 3; // 鼠标按键状态
}

message EndPointInfo {
  enum EndPointType {
    IS_CLIENT = 0;
    IS_SERVER = 1;
  }

  EndPointType type = 1;
  string ID = 2;
}

message ProtoMessage {
  enum DataType {
    VIDEO_MESSAGE = 0;
    MOUSE_MESSAGE = 1;
    ENDPOINT_INFO = 2;
  }

  DataType type = 1; // 第一个字节，表示数据类型

  oneof data {
    VideoMessage VideMessageI = 2; // 视频数据
    MouseMessage MouseMessageI = 3; // 鼠标数据
    EndPointInfo EndPointInfoI = 4;
  }
}

message EndPointInfo {
  enum EndPointType {
    UNKNOWN = 0;
    SERVER = 1;
    CLIENT = 2;
  }

  EndPointType type = 1;

  string id = 2;
}
